presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -422,17 +423,9 @@ private static Throwable failureCause(StageInfo stageInfo)
         @Override
         public SqlQueryExecution createQueryExecution(QueryId queryId, String query, Session session, Statement statement)
         {
-            int initialHashPartitions;
+            int initialHashPartitions = getHashPartitionCount(session, this.initialHashPartitions);
```

Comment:
So if both big query and this property are set it will use the big query settings, it seems more natural to me to have the reverse since this property is more specific than the big query setting.